yowsup/layers/network/layer.py

```diff
@@ -56,7 +56,9 @@ def handle_connect(self):
 
     def handle_close(self, reason = "Connection Closed"):
         logger.debug("Disconnected, reason: %s" % reason)
-        self.emitEvent(YowLayerEvent(self.__class__.EVENT_STATE_DISCONNECTED, reason = reason, detached=True))
+        ev = YowLayerEvent(self.__class__.EVENT_STATE_DISCONNECTED, reason = reason, detached=True)
+        self.emitEvent(ev)
+        self.getStack().broadcastEvent(ev)
```

Comment:
why is that ? You are sending the event twice this way, bottom-up, and up-bottom